rclcpp/CMakeLists.txt

```diff
@@ -387,6 +387,14 @@ if(BUILD_TESTING)
       "rcl")
     target_link_libraries(test_utilities ${PROJECT_NAME})
   endif()
+
+  ament_add_gtest(test_multi_threaded_executor test/executors/test_multi_threaded_executor.cpp
```

Comment:
I added an `#ifdef` around the corresponding sched_setscheduler(), because the bug originally manifested on OSX, so we wanted to test against that.  That trick additionally isn't necessary on OSX because the bug shows up with the default scheduler.